src/processor.cpp

```diff
@@ -0,0 +1,394 @@
+        //TODO EDIT: Verify in mapdata.cpp verification functions instead of on examine
+        debugmsg("Examined %s has action interact_with_processor, but has no processor data associated", g->m.furn(examp).id().c_str());
+        return;
+    }
+
+    const processor_data &current_processor = processors_data.obj(pid);
+    //Figure out active process
+    process_data active_process = select_active( examp );
+
+    //Msg to tell you what you need for a process
+    add_msg(_("You start on %s."), active_process.name.c_str());
+    for (const itype_id &i : active_process.components) {
+        add_msg(_("You need some %s for each."), item::nname(i, 1).c_str());
+    }
+    map_stack items = g->m.i_at(examp);
+    std::map<itype_id, long> charges;
+    item fuel = item::item(current_processor.fuel_type, int(calendar::turn), 0);
+    for ( item &i : items) {
+        //Please check to make sure this one is working as intended
+        auto index = std::find( active_process.components.begin(), active_process.components.end(), i.typeId() );
+        if (index == active_process.components.end() && i.typeId() != current_processor.fuel_type) {
+            add_msg(m_bad, _("You find some %s, which can't be used in this process."), i.tname(1, false).c_str());
+            add_msg(_("You must remove it before you proceed."));
+            return;
+        }
+        else if (i.typeId() != current_processor.fuel_type) {
+            charges.insert( std::make_pair( i.typeId() , charges.at( i.typeId() ) + i.charges ) );
+        }
+        else {
+            fuel.charges += i.charges;
+        }
+    }
+    //At this point we only have components and fuel on tile
+    //And we know exact numbers
+    long minimum = LONG_MAX;
+    for( std::pair<itype_id, long> m : charges ){
+        if( m.second == 0 ){
+            add_msg(m_info, _("There's no %s which is needed for the process"), item::nname( m.first, 1 ).c_str() );
+            return;
+        } else if( m.second < minimum ){
+            minimum = m.second;
+        }
+    }
+    //Need to check if the next line performs integer division
+    minimum = std::min( minimum, fuel.charges / active_process.fuel_intake );
+    if ( minimum <= 0 ) {
+        add_msg(_("The current materials present would go to waste without producing any %s, you need more reagents"), item::nname(active_process.output, 1).c_str() );
+        return;
+    }
+    add_msg( _("This process is ready to be started, and it will produce %d %s"), item::nname(active_process.output, minimum).c_str(), minimum );
+    if( !p.has_charges( "fire" , 1 ) ) {
+        add_msg( _("This process is ready to be started, but you have no fire source to start it.") );
+        return;
+    } else if( !query_yn( _("This process is ready to be started, and it will produce %d %s . %s"),
+                        item::nname(active_process.output, minimum).c_str(), minimum, current_processor.full_message.c_str() ) ) {
```

Comment:
You got order mixed up here - `minimum` should go before `nname` because in the format string above `%d` goes before `%s`.